cmake/warnings.cmake

```diff
@@ -20,6 +20,12 @@ endif ()
 
 option (WEVERYTHING "Enables -Weverything option with some exceptions. This is intended for exploration of new compiler warnings that may be found to be useful. Only makes sense for clang." ON)
 
+# Control maximum size of stack frames. It can be important if the code is run in fibers with small stack size.
+# Only in release build because debug has too large stack frames.
+if ((NOT CMAKE_BUILD_TYPE_UC STREQUAL "DEBUG") AND (NOT SANITIZE))
+    add_warning(frame-larger-than=16384)
```

Comment:
I'm seeing the following on GCC 10.2:

```
In file included from ../src/Core/Settings.h:3,
                 from ../src/Core/Settings.cpp:1:
../src/Core/Settings.cpp: In function 'DB::SettingsTraits::Accessor::instance()::<lambda()>':
../src/Core/BaseSettings.h:829:9: error: the frame size of 16416 bytes is larger than 16384 bytes [-Werror=frame-larger-than=]
  829 |         }(); \
      |         ^
../src/Core/Settings.cpp:20:1: note: in expansion of macro 'IMPLEMENT_SETTINGS_TRAITS'
   20 | IMPLEMENT_SETTINGS_TRAITS(SettingsTraits, LIST_OF_SETTINGS)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

Reply:
You can simply increase the limit two times.
If the error will happen in our CI, we will either increase the limit or rewrite the code.

Reply:
https://github.com/ClickHouse/ClickHouse/pull/14609 is going to take care of this.